numba/roc/hsaimpl.py

```diff
@@ -234,9 +235,10 @@ def hsail_atomic_add_tuple(context, builder, sig, args):
     return builder.atomic_rmw("add", ptr, val, ordering='monotonic')
 
 
-@lower('hsail.smem.alloc', types.UniTuple, types.Any)
+@lower(roc.shared.array, types.UniTuple, types.Any)
```

Comment:
I can't work out why there was never a lowering for `hsail.smem.alloc` with an integer argument. I've swapped like-for-like here, but I can't tell whether this is going to work. Does another lowering:

```python
@lower(roc.shared.array, types.IntegerLiteral, types.Any)
```

need adding, or is there something I've missed that always converts these into `UniTuple`s?

Secondly, if there are two integer literals of different value, I suspect the argument type might be `types.Tuple` - does this case need handling?

Reply:
I can't tell why this works neither. We will need to check this backend on real HW.

Reply:
And for `Tuple` vs `UniTuple`, you are right:

```
In [3]: types.UniTuple.from_types([types.IntegerLiteral(1), types.IntegerLiteral(2)])
Out[3]: Tuple(Literal[int](1), Literal[int](2))

In [4]: types.UniTuple.from_types([types.IntegerLiteral(1), types.IntegerLiteral(1)])
Out[4]: UniTuple(Literal[int](1) x 2)
```

will need a version for `Tuple`

Reply:
I'll try and get some hardware sorted locally to test this on.

Reply:
Hopefully this is now resolved.